### DISCUSSION TITLE
Can't get JK B2A8S20P HV: V11.XW SV: V11.262 to connect via bluetooth

### DISCUSSION THREAD

**Discussion by akash23anand** [syssi/esphome-jk-bms#506] [Q&A] [answered]
hi,  I have two JK B2A8S20P bms's.  The newer version HV: V11.XA SV: V11.48 connects to two different ESP32s no problem but the older version (HV: V11.XW SV: V11.262) can't connect.   The config file for both is the same except the MAC Address which is specific for the bms so it should work.   Is there anything else I can provide to help debug this?  Since I know the yaml is solid I didn't choose to share that but lmk if you need it

![IMG_2212](https://github.com/syssi/esphome-jk-bms/assets/54643272/7d5e5802-2f3b-4cca-9c00-23c9732f27b3)

[logs_jk-bms-battery-monitor-1_run.txt](https://github.com/user-attachments/files/15541356/logs_jk-bms-battery-monitor-1_run.txt)

**Comment by syssi** [comment]
Please make sure the BLE LED of the BMS blinks which indicates no BLE client is connected and it's broadcasting the BLE advertisement. The BLE tracker component should report at the ESPHome log the advertisements of all BLE devices around. The log capture doesn't include any advertisements. That's pretty strange.

**Reply by akash23anand** [reply]
### I'm not sure this version of BMS has a BLE LED.  But in any case, I'm able to connect and disconnect to it via my iphone app so I know its possible and available to connect via bluetooth.  Can you think of anything else?  

This is the version I have, except the older version HV: V11.XW SV: V11.262
https://www.aliexpress.us/item/3256804134737264.html?spm=a2g0o.order_list.order_list_main.10.75cf18020TgtcD&gatewayAdapt=glo2usa

**Reply by syssi** [reply]
The BMS can handle a single connection only. Please make sure no other client/app is connected to the BMS. If there is no connection established the BMS will start advertising the service. The ESP must receive this advertisement. Without this broadcast the BLE client connection cannot be established.

**Reply by syssi** [reply]
Could you provide some more logs? What's the distance between the ESP and the BMS? Could you try to play around with the distance (go a few meter away vs. get closer).

**Reply by akash23anand** [reply]
I’m positive when I try to connect to the old bms nothing else is connected to it.  Currently the esp is within 1 meter to the bms but I can make it farther and closer and send you the logs tomorrow.  I’ll also turn off the new bms so that doesn’t interfere, although it probably doesn’t since that one is already connected via ble to another esp32.  Will share more tomorrow after I try the various placements

**Comment by akash23anand** [comment]
Unfortunately no change both far or near (like 3 inches) away with the newer version JK BMS turned off as to not interfere.  Here's my yaml in case you can spot anything on it but like I said this will connect with the new JK BMS when I swap the MAC address so I know it works.  Also added the logs which from my noob read doesn't explain much.  
[logs_jk-bms-battery-monitor-1_logs.txt](https://github.com/user-attachments/files/15598756/logs_jk-bms-battery-monitor-1_logs.txt)


https://pastebin.com/FDcUtkDk

**Reply by syssi** [reply]
Could you change the log section to:

```
logger:
  level: VERY_VERBOSE
  logs:
    esp32_ble_tracker: VERY_VERBOSE
    jk_bms_ble: VERY_VERBOSE
    scheduler: DEBUG
    component: DEBUG
    sensor: DEBUG
    mqtt: INFO
    mqtt.idf: INFO
    mqtt.component: INFO
    mqtt.sensor: INFO
    mqtt.switch: INFO
    api.service: INFO
    api: INFO
```

and provide another log?

**Reply by syssi** [reply]
```
[19:07:00][W][esp32_ble_tracker:110]: Too many BLE events to process. Some devices may not show up.
```

Do you have a lot of BLE devices around?

**Reply by akash23anand** [reply, marked as answer]
Sorry this was complete my fault.  For some reason under the normal logger: Debug  I assumed the 'other' mac address was the old BMS as there were only two devices being advertised.  However with the verbose logger i was able to see the names for each of the addresses and found out that the old BMS had a different mac address, once updated it worked via the web server, but no HA.  So I put the old debug level logger back on and it works perfectly.  My Wyze hub was blowing this thing up and had like 7 different records in the ble tracker for the same mac address so maybe that was causing it confusion.  That may be an enhancement you consider for the BLE advertiser is to increase the amount of devices it can look at but also include names.  If that's already there and some how I missed it, my apologies.  Thanks again for your help in getting me setup, I love this integration!
